src/Microsoft.AspNet.Mvc.TagHelpers/ValidationMessageTagHelper.cs

```diff
@@ -47,10 +47,12 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 {
                     output.MergeAttributes(tagBuilder);
 
+                    var childContent = await context.GetChildContentAsync();
+
                     // We check for whitespace to detect scenarios such as:
                     // <span validation-for="Name">
                     // </span>
-                    if (string.IsNullOrWhiteSpace(output.Content))
+                    if (string.IsNullOrWhiteSpace(childContent) && string.IsNullOrEmpty(output.Content))
```

Comment:
as in `<label/>` helper, be expllcit and always set `Content` when it's currently `null` or empty.